x-pack/plugins/osquery/public/action_results/action_agents_status_bar.tsx

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import styled from 'styled-components';
+import { EuiColorPaletteDisplay } from '@elastic/eui';
+import React, { useMemo } from 'react';
+
+import { AGENT_STATUSES, getColorForAgentStatus } from './services/agent_status';
+import type { ActionAgentStatus } from './types';
+
+const StyledEuiColorPaletteDisplay = styled(EuiColorPaletteDisplay)`
+  &.osquery-action-agent-status-bar {
+    border: none;
+    border-radius: 0;
+    &:after {
+      border: none;
+    }
+  }
+`;
+
+export const AgentStatusBar: React.FC<{
+  agentStatus: { [k in ActionAgentStatus]: number };
+}> = ({ agentStatus }) => {
+  const palette = useMemo(() => {
+    return AGENT_STATUSES.reduce((acc, status) => {
+      const previousStop = acc.length > 0 ? acc[acc.length - 1].stop : 0;
```

Comment:
i think this could be more readable if rewritten like so:
```
 let stop = 0;
 return AGENT_STATUSES.reduce((acc, status) => {
      acc.push({
        stop,
        color: getColorForAgentStatus(status),
      });
      stop += agentStatus[status] || 0
      return acc;
    }, [] as Array<{ stop: number; color: string }>);
  }, [agentStatus]);
```